src/generic/AhoCorasick.jl

```diff
@@ -0,0 +1,201 @@
+#
+#   FreeAssAhoCorasick.jl : implement bulk divide check for leading terms of free associative Algebra elements
+#   for use e.g. in Groebner Basis computation
+#
+###############################################################################
+
+
+export search,
+    AhoCorasickAutomaton, insert_keyword!, aho_corasick_automaton, AhoCorasickMatch#, Word
+
+import DataStructures.Queue, DataStructures.enqueue!, DataStructures.dequeue!
+
+const Word = Vector{Int}
+
+Markdown.@doc doc"""
+    AhoCorasickAutomaton
+
+An Aho-Corasick automaton, which can be used to efficiently search for a fixed list of keywords (vectors of Ints) in
+arbitrary lists of integers
+
+# Examples:
+```jldoctest; setup = :(using AbstractAlgebra.Generic)
+julia> keywords = [[1, 2, 3, 4], [1, 5, 4], [4, 1, 2], [1, 2]]
+julia> aut = aho_corasick_automaton(keywords)
+julia> search(aut, [10, 4, 1, 2, 3, 4])
+AhoCorasickMatch(6, 1, [1, 2, 3, 4])
+
+
```

Comment:
There must be an empty line between the inputs. 
```suggestion
julia> keywords = [[1, 2, 3, 4], [1, 5, 4], [4, 1, 2], [1, 2]];

julia> aut = aho_corasick_automaton(keywords);

julia> search(aut, [10, 4, 1, 2, 3, 4])
AhoCorasickMatch(6, 1, [1, 2, 3, 4])
```